src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
+                throw new Error('Invalid elicitation Action');
+            }
         } else {
-            this.state.isValidValue = true;
-            this.state.lastInitiative.actName = ConfirmValueAct.name;
+            this.state.confirmed = false;
             resultBuilder.addAct(
-                new ConfirmValueAct(this, {
+                new InvalidValueAct<number>(this, {
                     value: this.state.value,
-                    renderedValue:
-                        this.state.value !== undefined
-                            ? this.props.valueRenderer(this.state.value, input)
-                            : '',
+                    renderedValue: this.props.valueRenderer(this.state.value!, input),
+                    reasonCode: validationResult.reasonCode,
+                    renderedReason: validationResult.renderedReason,
                 }),
             );
+            this.askElicitationQuestion(input, resultBuilder, elicitationAction);
         }
     }
 
-    /* TODO: bug: User's action has to be used in this function to form the response
-     * prompt, need to define action in the related payloadType This will fix
-     * the following: U: No, {change} the value please A: What value do you want
-     * to {change} to?
-     */
-    private async commonHandlerWhenValueRejected(
+    private askElicitationQuestion(
         input: ControlInput,
         resultBuilder: ControlResultBuilder,
-    ): Promise<void> {
-        this.state.rejectedValues.push(this.state.value!);
-        const ambiguousPartner = this.getAmbiguousPartner(this.state.value!);
-        if (ambiguousPartner !== undefined && !this.state.rejectedValues.includes(ambiguousPartner)) {
-            const previousValue = this.state.value;
-            this.state.value = ambiguousPartner;
-            const validationResult = await evaluateValidationProp(this.props.validation, this.state, input);
-            if (validationResult === true) {
-                // this is to confirm from users for the suggestedValue
-                this.state.isValidValue = true;
-                this.state.lastInitiative.actName = ConfirmValueAct.name;
-                resultBuilder.addAct(
-                    new SuggestValueAct(this, {
-                        value: this.state.value,
-                        renderedValue:
-                            this.state.value !== undefined
-                                ? this.props.valueRenderer(this.state.value, input)
-                                : '',
-                    }),
-                );
-            } else {
-                this.state.value = previousValue;
-                resultBuilder.addAct(new RequestValueAct(this));
+        elicitationAction: string,
+    ) {
+        this.state.elicitationAction = elicitationAction;
+        this.addInitiativeAct(new RequestValueAct(this), resultBuilder);
+    }
+
+    addInitiativeAct(initiativeAct: InitiativeAct, resultBuilder: ControlResultBuilder) {
```

Comment:
This is only sometimes used, but not always. eg confirmValue @ :1042 doesn't use this method.  It would be better to be consistent: either always call this, or remove this method.